core/src/main/java/org/opensearch/sql/planner/logical/LogicalSort.java

```diff
@@ -20,11 +20,14 @@
 @EqualsAndHashCode(callSuper = true)
 public class LogicalSort extends LogicalPlan {
 
+  private final Integer count;
```

Comment:
Add javadoc.

Reply:
I would prefer if we made this an `Optional<Integer>` instead of having the nullability be implicit

This makes the contract obvious to anyone using/reading the class. We can also then have the constructor guard against non-positive integers by initializing it as `None`.